android/src/ti/bluetooth/TiBluetoothModule.java

```diff
@@ -8,301 +8,251 @@
+
+	@Kroll.constant
+	public static final int TYPE_L2CAP = BluetoothSocket.TYPE_L2CAP;
+	@Kroll.constant
+	public static final int TYPE_RFCOMM = BluetoothSocket.TYPE_RFCOMM;
+	@Kroll.constant
+	public static final int TYPE_SCO = BluetoothSocket.TYPE_SCO;
+
+	public final int DEFAULT_SCAN_MODE = SCAN_MODE_BALANCED;
+	private int scanmode = DEFAULT_SCAN_MODE;
+
+	public TiBluetoothModule() {
+		super();
+		appContext = TiApplication.getInstance();
+		activity = appContext.getCurrentActivity();
+		appContext.registerReceiver(new BlutoothStateChangedBroadcastReceiver(
+				TiBluetoothModule.this, btAdapter), new IntentFilter(
+				BluetoothAdapter.ACTION_STATE_CHANGED));
+	}
+
+	@Kroll.onAppCreate
+	public static void onAppCreate(TiApplication app) {
+		Log.d(LCAT, "inside onAppCreate");
+	}
+
+	private final ScanCallback scanCallback = new ScanCallback() {
+		@Override
+		public void onScanResult(int callbackType, ScanResult result) {
+			BluetoothDevice device = result.getDevice();
+			if (device != null) {
+				BluetoothDeviceProxy btDeviceProxy = new BluetoothDeviceProxy(
+						device);
+				// TODO resultpayload is array of btDeviceProxy's
+				Log.d(LCAT, "Found something " + device.getName());
+				if (device.getName() != null) {
+					Log.d(LCAT,
+							"Found: " + device.getName() + " "
+									+ device.getAddress());
+
+					ArrayList<String> ids = new ArrayList<String>();
+					for (ParcelUuid id : device.getUuids()) {
+						ids.add(id.toString());
+					}
+					KrollDict kd = new KrollDict();
+					kd.put("device", btDeviceProxy);
+
+					kd.put("name", device.getName());
+					kd.put("address", device.getAddress());
+					kd.put("bondState", device.getBondState());
+					kd.put("ids", ids.toArray());
+					fireEvent("didDiscoverPeripheral", kd);
+					BluetoothGatt bluetoothGatt = device.connectGatt(
+							appContext, false,
+							new BluetoothGattCallbackHandler(
+									TiBluetoothModule.this));
+					btScanner.stopScan(scanCallback);
+				}
+			}
+		}
+	};
+
+	private List<ScanFilter> scanFilters() {
+		String[] ids = {};
+		return scanFilters(ids);
+	}
+
+	private List<ScanFilter> scanFilters(String[] ids) {
+		List<ScanFilter> list = new ArrayList<ScanFilter>(1);
+		for (int i = 0; i < ids.length; i++) {
+			ScanFilter filter = new ScanFilter.Builder().setServiceUuid(
+					ParcelUuid.fromString(ids[i])).build();
+			list.add(filter);
+		}
+		return list;
+	}
+
+	@Kroll.method
+	public int getState() {
+		return currentState;
+	}
+
+	@Kroll.method
+	public boolean isScanning() {
+		return isScanning;
+	}
+
+	@Kroll.method
+	public void initialize() {
+		btManager = (BluetoothManager) appContext
+				.getSystemService(Context.BLUETOOTH_SERVICE);
+		btAdapter = btManager.getAdapter();
+		if (btAdapter != null) {
+			setCurrentState(btAdapter.getState());
+		} else {
+			setCurrentState(MANAGER_STATE_UNSUPPORTED);
+		}
+	}
+
+	@Kroll.method
+	public void startScan() {
```

Comment:
Example brings an error that `startScan()` with 0 parameters is not available. You can change the next startScan method and add `@Kroll.argument(optional=true) int _scanmode` which should work